### PR TITLE
Add Debugging subheading to Install CLI page

### PR DESCRIPTION
**Pull request by benjaminapetersen**
TODO:
- [x] Lets change this to a dedicated debugging page
    - [x] Move the env var to this page
    - [x] Add the debugging yaml to this page
    - [x] Add the `rm -rf ~/.config/pinniped` trick to this page, for "logging out", clearing caches, etc.

**Release note**:

```release-note

```

### CODE REVIEW

**Comment by cfryanr** [comment]
Good idea to document this. I think it might make more sense at the bottom of this page https://pinniped.dev/docs/howto/login/ which is where we show how to use the `pinniped` CLI.

**Comment by benjaminapetersen** [comment]
I'd even consider an additional "Debugging" page where we document this env var as well as the YAML changes to enable debugging for server side components. 

Screenshot 2024-01-02 at 4 30 24 PM

**Comment by cfryanr** [comment]
> I'd even consider an additional "Debugging" page where we document this env var as well as the YAML changes to enable debugging for server side components.

That sounds reasonable to me.

File: site/content/docs/howto/debugging.md [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+```
+Then apply your configuration.
+
+The `log_level` options are as follows: 
+- `info` ("nice to know" information) 
+- `debug` (developer information) 
+- `trace` (timing information)
+- `all` (kitchen sink). 
+
+Do not use `trace` or all on production systems, as credentials may get logged. When this value is left unset,
```
Comment (cfryanr) [inline site/content/docs/howto/debugging.md:35]:
```suggestion
Do not use `trace` or `all` on production systems, as credentials may get logged. When this value is left unset, 
```

File: site/content/docs/howto/debugging.md [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+## Debugging on the Server
+
+To adjust the log level of the server side components of Pinniped (such as the Supervisor, Concierge,
+Impersonation Proxy, etc) edit the `log_level` configuration in your local copy of the Pinniped GitHub 
+repository. Open the `deploy/supervisor/values.yaml` or `deploy/concierge/values.yaml` file and edit: 
+
+```yaml
+log_level: "info|debug|trace|all"
+```
+Then apply your configuration.
```
Comment (cfryanr) [inline site/content/docs/howto/debugging.md:27]: Log level only applies to the Supervisor and Concierge.

These instructions assume that you are going to use ytt to deploy, but we actually document several ways to install the Supervisor and the Concierge (ytt or use yaml attached to our GitHub releases). This should describe both, or describe where to edit the value live on the cluster. Note that if you edit the value live on the cluster then you also need to restart the pods.

File: site/content/docs/howto/debugging.md [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+menu:
+  docs:
+    name: Debugging
+    weight: 900
+    parent: howtos
+---
+
+## Debugging on the Client 
+
+The `PINNIPED_DEBUG=true` environment variable can be set to enable additional CLI logging.
```
Comment (joshuatcasey) [inline site/content/docs/howto/debugging.md:16]: :)

File: site/content/docs/howto/debugging.md (hunk 2 above)
Comment (joshuatcasey) [reply site/content/docs/howto/debugging.md:27]: Makes sense, the user should know what value in the configmap is being updated by this change.

**Review by joshuatcasey — APPROVED** [review]
lgtm, just add the note about how to edit the log level inside the configmap

File: site/content/docs/howto/debugging.md (hunk 1 above)
Comment (benjaminapetersen) [reply site/content/docs/howto/debugging.md:35]: 👍

File: site/content/docs/howto/debugging.md (hunk 2 above)
Comment (benjaminapetersen) [reply site/content/docs/howto/debugging.md:27]: Agree, updating.

**Review by joshuatcasey — APPROVED** [review]

File: site/content/docs/howto/debugging.md [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+Do not use `trace` or `all` on production systems, as credentials may get logged. When this value is left unset, 
+only warnings and errors are printed. There is no way to suppress warning and error logs.
+
+Choose an update method that corresponds with the original installation method you chose for your cluster. Consult
+[Install Supervisor]({{< ref "../howto/install-supervisor" >}}) or [Install Concierge]({{< ref "../howto/install-concierge" >}})
+for more information.
+
+### Using kapp
+
+To adjust the log level of the Pinniped Supervisor or Concierge using `kapp`, edit the `log_level` in the `values.yaml` files
```
Comment (cfryanr) [inline site/content/docs/howto/debugging.md:38]: I think you meant to say using `ytt`, not using `kapp`.

And you don't edit the values.yaml file. See the other docs for instructions.

File: site/content/docs/howto/debugging.md [hunk 5]
```diff
@@ -0,0 +1,101 @@
...
+Then apply your configuration via `kapp deploy -f`.
+
+### Using kubectl
+
+To adjust the log level of the Pinniped Supervisor or Concierge using `kubectl`, find the configmap on your cluster within
+the namespace of each deployment:
+
+```bash
+# get the concierge config
+kubectl get cm pinniped-concierge-config --namespace concierge --output yaml > cm.concierge.yaml
```
Comment (cfryanr) [inline site/content/docs/howto/debugging.md:54]: Does this have the word `static` in its name, or is it really different from the supervisor?

File: site/content/docs/howto/debugging.md [hunk 6]
```diff
@@ -0,0 +1,101 @@
...
+  namespace: supervisor
+data:
+  pinniped.yaml: |
+    apiGroupSuffix: pinniped.dev
+    log:
+      level: "info|debug|trace|all"
+    # ...
+```
+
+And then apply  your configuration via `kubectl apply -f`.
```
Comment (cfryanr) [inline site/content/docs/howto/debugging.md:89]: After applying the configuration, you must also restart the pods for the change to get picked up.

This whole section might make more sense if you split it into two sections: one for updating the log level for the Supervisor and another for updating the log level for the Concierge. The commands are slightly different, and the file that you show above would actually not have both configmaps in it if you use the commands that you show above to create the files.

File: site/content/docs/howto/debugging.md (hunk 5 above)
Comment (benjaminapetersen) [reply site/content/docs/howto/debugging.md:54]: It does have the word `-static` in the name:



They are unfortunately slightly different in name.

File: site/content/docs/howto/debugging.md (hunk 4 above)
Comment (benjaminapetersen) [reply site/content/docs/howto/debugging.md:38]: Ah, yeah I see we tell users to create a NEW `values.yaml` file and add just the config they desire to change:

```yaml
# ./my-supervisor-values.yaml

#@data/values
---
image_tag: v0.29.0
log_level: debug
replicas: 1
```
And then something `ytt --file /path/to/pinniped/deploy/supervisor --file ./my-supervisor-values.yaml`

File: site/content/docs/howto/debugging.md [hunk 7]
```diff
@@ -0,0 +1,101 @@
...
+the namespace of each deployment:
+
+```bash
+# get the concierge config
+kubectl get cm pinniped-concierge-config --namespace concierge --output yaml > cm.concierge.yaml
+# get the supervisor config
+kubectl get cm pinniped-supervisor-static-config --namespace supervisor --output yaml > cm.supervisor.yaml
+```
+
+Edit the yaml files:
```
Comment (cfryanr) [inline site/content/docs/howto/debugging.md:59]: `Edit the yaml files to change the log level:`

**Pull request merged**